File: mkdir_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def mkdir_dir(dir_path, config):
    if not check_file_permission(dir_path, "write", config):
        raise PermissionError(f"Write access denied for {dir_path}")
    dir_path = Path(dir_path).resolve()
    dir_path.mkdir(parents=True, exist_ok=True)
    return f"Created directory {dir_path}"

File: ts-tools/mkdir_tool.ts
import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function mkdirDir(dirPath: string, config: any): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(dirPath, 'write', config)
      .then(() => {
        const resolvedPath = path.resolve(dirPath);
        fs.mkdir(resolvedPath, { recursive: true }, (err) => {
          if (err) {
            reject(err);
          } else {
            resolve(`Created directory ${dirPath}`);
          }
        });
      })
      .catch(reject);
  });
}
